modulemd/test-modulemd-defaults.c

```diff
@@ -393,6 +393,331 @@ modulemd_defaults_test_missing_default_from_profile (DefaultsFixture *fixture,
 }
 
 
+static void
+modulemd_defaults_test_copy (DefaultsFixture *fixture, gconstpointer user_data)
+{
+  g_autofree gchar *yaml_path = NULL;
+  GError *error = NULL;
+  g_autoptr (ModulemdDefaults) orig = NULL;
+  g_autoptr (ModulemdDefaults) copy = NULL;
+
+  yaml_path = g_strdup_printf ("%s/mod-defaults/spec.v1.yaml",
+                               g_getenv ("MESON_SOURCE_ROOT"));
+  g_assert_nonnull (yaml_path);
+  orig = modulemd_defaults_new_from_file (yaml_path, &error);
+
+  copy = modulemd_defaults_copy (orig);
+  g_assert_nonnull (copy);
+
+  g_assert_cmpstr (modulemd_defaults_peek_module_name (orig),
+                   ==,
+                   modulemd_defaults_peek_module_name (copy));
+
+  g_assert_cmpstr (modulemd_defaults_peek_default_stream (orig),
+                   ==,
+                   modulemd_defaults_peek_default_stream (copy));
+
+  g_assert_cmpint (
+    g_hash_table_size (modulemd_defaults_peek_profile_defaults (orig)),
+    ==,
+    g_hash_table_size (modulemd_defaults_peek_profile_defaults (copy)));
+}
+
+
+static void
+modulemd_defaults_test_merging (DefaultsFixture *fixture,
+                                gconstpointer user_data)
+{
+  g_autofree gchar *yaml_path = NULL;
+  g_autoptr (GPtrArray) objects = NULL;
+  g_autoptr (GPtrArray) override_objects = NULL;
+  g_autoptr (GPtrArray) merged_base = NULL;
+  g_autoptr (GPtrArray) overridden = NULL;
+  ModulemdDefaults *defaults = NULL;
+  GError *error = NULL;
+
+  yaml_path = g_strdup_printf ("%s/test_data/defaults/merging-base.yaml",
+                               g_getenv ("MESON_SOURCE_ROOT"));
+  g_assert_nonnull (yaml_path);
+
+  objects = modulemd_objects_from_file (yaml_path, &error);
+  g_clear_pointer (&yaml_path, g_free);
+  g_assert_nonnull (objects);
+  g_assert_cmpint (objects->len, ==, 6);
+
+  merged_base = modulemd_merge_defaults (objects, NULL, FALSE, &error);
+  if (!merged_base)
+    {
+      fprintf (stderr, "Error merging defaults: %s\n", error->message);
```

Comment:
g_warning? or you can actually even use g_error.

Reply:
I intentionally didn't use those because they terminate immediately. I meant to go back and test that the error->code and quark match as well and forgot to do that. I'll fix that up later.